src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -164,31 +167,49 @@ public AnomalyDetector(
         User user
     ) {
         if (Strings.isBlank(name)) {
-            throw new IllegalArgumentException("Detector name should be set");
+            throw new ADValidationException("Detector name should be set", DetectorValidationIssueType.NAME, ValidationAspect.DETECTOR);
```

Comment:
AD frontend limit detector name "Detector name must contain 1-64 characters. Valid characters are a-z, A-Z, 0-9, -(hyphen) and _(underscore).". 
Feature name has the same limitation on frontend.

How about we add the same limitation at backend ?